File: src/DeployManager.Client/src/shared/types/deployJob.ts
export interface DeployJob {
  id: string
  siteId: string
  siteName: string
  environmentName: string
  fileName: string
  fileSize: number
  status: string
  logSummary: string | null
  errorMessage: string | null
  startedAt: string | null
  completedAt: string | null
  createdAt: string
  createdByUsername: string | null
}

export interface DeployJobDetail extends DeployJob {
  siteCode: string
  targetType: string
  hasBackup: boolean
  logs: DeployLogEntry[]
}

export interface DeployLogEntry {
  timestamp: string
  level: string
  message: string
}

export interface DeployJobsQueryParams {
  status?: string
  from?: string
  to?: string
  siteId?: string
  environmentId?: string
}

File: src/DeployManager.Domain/Entities/DeployJob.cs
using DeployManager.Domain.Common;
using DeployManager.Domain.Enums;

namespace DeployManager.Domain.Entities;

public class DeployJob : BaseEntity
{
    public Guid SiteId { get; set; }
    public Guid? PackageId { get; set; }
    public string FileName { get; set; } = string.Empty;
    public long FileSize { get; set; }
    public DeployStatus Status { get; set; } = DeployStatus.Pending;
    public string? LogSummary { get; set; }
    public string? ErrorMessage { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public Guid? CreatedByUserId { get; set; }
    public DeploySite? Site { get; set; }
    public DeployPackage? Package { get; set; }
    public User? CreatedByUser { get; set; }
    public ICollection<DeployLog> Logs { get; set; } = new List<DeployLog>();
    public ICollection<Backup> Backups { get; set; } = new List<Backup>();
}
